readme.md
# Stream Overlays using PHP/HTML Etcetera

These are stream overlays that I am using.

They are using PHP as a server side programming language.

They use HTML for markup.

Currently no other services are being used.



changelog.md
Version 0.0.1

- Initial Release
- Based on previous version
- Added .travis.yml 



intermission/index.php
<?php 

$refresh=rand(1,20); 

header("Refresh:$refresh");

include_once("intermissionstart.php");
include_once("timefunctions.php");
include_once("defaults.php");

$now=time();

$intermissionStart=intermissionStartTime($startfile);
$elapsedTime=$now-$intermissionStart;

$intermissionDurration=intermissionDurration($durrfile);
$intermissionTimeLeft=$intermissionDurration-$elapsedTime;

$intermissionReason=intermissionReason($reasonfile);

$startString=date("D Y F j H:i:s", $intermissionStart);
$nowString=date("D Y F j H:i:s", $now);
$durrString=time_elapsed_A($intermissionDurration);
$elapsedString=time_elapsed_A($elapsedTime);
$leftString=time_elapsed_A($intermissionTimeLeft);
$reasonString=$intermissionReason;


?>
<!doctype html>
<html>
<head>
	<title>Stream Intermission</title>
	<meta http-equiv="refresh" content="<?php echo $refresh; ?>">
</head>
<body>
	<h1>Stream is on Intermission</h1>
	<p>This stream is currently on intermission</p>
	<p>Sadly, this intermission page isn't setup correctly, so you get limited information</p>
	<h2>Intermission Details</h2>
	<ul>
		<li>Intermission Reason: <?php echo $reasonString; ?></li>
		<li>Intermission Start: <?php echo $startString; ?></li>
		<li>Current Time: <?php echo $nowString; ?></li>
		<li>Intermission Durration: <?php echo $durrString; ?></li>
		<li>Time Elapsed: <?php echo $elapsedString; ?></li>
		<li>Expected Return: <?php echo $leftString; ?></li>
		<li>Refreshing in: <?php echo $refresh; ?> seconds</li>
	</ul>
</body>
</html>


intermission/timefunctions.php
<?php

function time_elapsed_A($secs){
	if ($secs > 0) {
		$sign="+";
	} elseif ($secs < 0) {
		$sign="-";
	} else {
		$sign="0";
	}

//	echo "sign: ".$sign;

	if ($sign=="-") {
		$secs = $secs*-1;
		$pre= "-(";
		$post=")";
//		echo $sign;
	} elseif ($sign=="+") {
		$pre = "";
		$post = "";
//		echo $sign;
	} elseif ($sign=="0") {
		$pre = "0";
		$post = "s";
//		echo $sign;
	} else {
		$pre = "oops";
		$post = "something went wrong";
//		echo $sign;
	}



    $bit = array(
        'y' => $secs / 31556926 % 12,
        'w' => $secs / 604800 % 52,
        'd' => $secs / 86400 % 7,
        'h' => $secs / 3600 % 24,
        'm' => $secs / 60 % 60,
        's' => $secs % 60
        );
        
    foreach($bit as $k => $v) {
        if($v > 0)$ret[] = $v . $k;
	}

//	echo "pre: ".$pre;
//	echo "post: ".$post;

//	print_r($ret);

	$result = $pre;
	if (isset($ret) && is_array($ret)) { $result .= join(' ', $ret); }
	$result .= $post;

    return $result;
    }
    

function time_elapsed_B($secs){
    $bit = array(
        ' year'        => $secs / 31556926 % 12,
        ' week'        => $secs / 604800 % 52,
        ' day'        => $secs / 86400 % 7,
        ' hour'        => $secs / 3600 % 24,
        ' minute'    => $secs / 60 % 60,
        ' second'    => $secs % 60
        );
        
    foreach($bit as $k => $v){
        if($v > 1)$ret[] = $v . $k . 's';
        if($v == 1)$ret[] = $v . $k;
        }
    array_splice($ret, count($ret)-1, 0, 'and');
    $ret[] = '.';
    
    return join(' ', $ret);
    }
    

    
    

/** Output:
time_elapsed_A: 6d 15h 48m 19s
time_elapsed_B: 6 days 15 hours 48 minutes and 19 seconds ago.
**/
?>


intermission/intermissionstart.php
<?php 
/*********
 * Intermission Start
 *
 * This sets the variable for the start of the intermission
 *
 ********/

$filename="intermissionstart.date";

function intermissionStartTime($filename) {
	if (file_exists($filename) && is_readable($filename)) {
		$file=fopen($filename, "r");
		$startTime=fgets($file);
		if (date("D Y F j, H:i:s", $startTime)) {
			return $startTime;
		} else {
			return FALSE;
		}
	} else {
		return FALSE;
	}
}

function startIntermission($startname, $durrationname, $reasonname, $durration, $reason) {
	if (file_exists($startname) && is_writable($startname)) {
                $file=fopen($startname, "w");
		$startTime=time();
                fwrite($file, $startTime);
		fclose($file);
	}
	if (file_exists($durrationname) && is_writable($durrationname)) {
                $file=fopen($durrationname, "w");
                fwrite($file, $durration);
		fclose($file);
	}
	if (file_exists($reasonname) && is_writable($reasonname)) {
                $file=fopen($reasonname, "w");
                fwrite($file, $reason);
                fclose($file);
	}
	
}       

function intermissionDurration($filename) {
	if (file_exists($filename) && is_readable($filename)) {
		$file=fopen($filename, "r");
		$startTime=fgets($file);
		if (date("D Y F j, H:i:s", $startTime)) {
			return $startTime;
		} else {
			return FALSE;
		}
	} else {
		return FALSE;
	}
}

function intermissionReason($reasonname) {
	if (file_exists($reasonname) && is_readable($reasonname)) {
		$file=fopen($reasonname, "r");
		$reason=fgets($file);
		return $reason;
	} else {
		return FALSE;
	}
}



intermission/startintermission.php
<?php 

include_once("intermissionstart.php");
include_once("defaults.php");
include_once("timefunctions.php");

$multiplier = array(
	"seconds" => 1,
	"minutes" => 60,
	"hours" => 3600,
	"days" => 3600*24,
	"months" => 3600*24*30.5,
	"years" => 3600*24*30.5*12
);

if (isset($_POST["value"]) && isset($_POST["units"])) {
	$value=$_POST["value"];
	$units=$_POST["units"];
	$reason=$_POST["reason"];
	$durration=$value*$multiplier[$units]+rand(0,$multiplier[$units]);
	startIntermission($startfile, $durrfile, $reasonfile, $durration, $reason);
}

$self=$_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];

$now=time();

$intermissionStart=intermissionStartTime($startfile);

$elapsedTime=$now-$intermissionStart;

$intermissionDurration=intermissionDurration($durrfile);

$intermissionReason=intermissionReason($reasonfile);

$intermissionTimeLeft=$intermissionDurration-$elapsedTime;

$startString=date("D Y F j H:i:s", $intermissionStart);
$nowString=date("D Y F j H:i:s", $now);
$durrString=time_elapsed_A($intermissionDurration);
$elapsedString=time_elapsed_A($elapsedTime);
$leftString=time_elapsed_A($intermissionTimeLeft);
$reasonString=$intermissionReason;


?>
<!DOCTYPE html>
<html>
<head>
	<title>Set the new intermissioin</title>
</head>
<body>
	<h1>Set the new intermission</h1>
	<p>This will set the new intermission</p>

	<form action="<?php $self ?>" method="post">
		<label for="reason">Reason:</label>
		<input id="reason" name="reason" type="text">
		<label for="value">Value:</label>
		<input id="value" type="text" name="value">
		<label for="units">Units:</label>
		<select id="units" name="units">
			<option value="seconds">Seconds</option>
			<option value="minutes">Minutes</option>
			<option value="hours">Hours</option>
			<option value="days">Days</option>
			<option value="months">Months</option>
			<option value="years">Years</option>
		</select>
		<input type="submit" value="Start Intermission">
	</form>

	<p>This is the current intermission:</p>
	<ul>
		<li>Intermission Reason: <?php echo $reasonString ?></li>
                <li>Intermission Start: <?php echo $startString; ?></li>
                <li>Current Time: <?php echo $nowString; ?></li>
                <li>Intermission Durration: <?php echo $durrString; ?></li>
                <li>Time Elapsed: <?php echo $elapsedString; ?></li>
                <li>Expected Return: <?php echo $leftString; ?></li>
        </ul>

</body>
</html>
